src/formats/sprite.c

```diff
@@ -310,6 +313,9 @@ int sd_sprite_vga_decode(sd_vga_image *dst, const sd_sprite *src) {
                     uint8_t b = src->data[i];
                     int pos = ((y * src->width) + x);
                     dst->data[pos] = b;
+                    if (b == transparent) {
+                        assert(false);
+                    }
```

Comment:
assert(b != transparent)

Reply:
That would add a function call for each iteration, not great for perf.

Reply:
assert is automatically stripped out in release builds. It's safe to use performance-wise.

Reply:
I don't want to impact the debug builds too much either.

Reply:
assert is a macro. This will (in theory) leave a condition to all release builds, which will slow them down. If you put the condition inside the assert macro, it will get removed and the speed will not change to either direction in debug builds.